### ISSUE TITLE
Formatting midnight for en-GB

### ISSUE THREAD

**Issue by jpresley23** [andyearnshaw/Intl.js#173]
After updating to version 1.20, I still get the following error when running tests with PhantomJS

> PhantomJS 2.1.1 (Mac OS X 0.0.0) TimeRangeSlider when the locale changes to GB, the time below  start time shows as "00::00" FAILED
        expected '0:00' to equal '00:00'

**Comment by caridy** [comment]
@jpresley23 can you provide more details? maybe the actual code that is producing the wrong value.

**Comment by caridy** [comment]
Also, is this a new issue? or an existing issue?

**Comment by jpresley23** [comment]
It's an issue that existed prior to 1.2, but it was not fixed by 1.2. The code for the test is below. We're using karma. Just as an fyi, I'm updating our build to use PhantomJS so I'm not as up to date with the code. The test passes with karma with Chrome and Firefox launchers.
```
import { expect } from 'chai';
import React from 'react';
import { IntlProvider } from 'react-intl';
import TestUtils from 'react-addons-test-utils';
import TimeRangeSlider from '../forms/TimeRangeSlider';
import i18n from '../Reacti18n';

describe('TimeRangeSlider', () => {
  let rootComponent;
  let renderedComponent;
  const mockFunction = () => {};

  beforeEach(() => {
    rootComponent = TestUtils.renderIntoDocument(
      <IntlProvider locale={i18n.getLocale()}>
        <TimeRangeSlider
          startTimeMin={0}
          startTimeMax={1439}
          handleFilterChange={mockFunction}
          ref={(c) => {
            renderedComponent = c.refs.wrappedInstance;
            return renderedComponent;
          }
        }
        />
      </IntlProvider>
    );
  });

  it('shows the select wrapper ', () => {
    const slider = TestUtils.findRenderedComponentWithType(
      rootComponent,
      TimeRangeSlider
    );
    return expect(slider).to.exist;
  });

  it('has its initial state set correctly', () => {
    expect(renderedComponent.props.startTimeMin).to.equal(0);
    expect(renderedComponent.props.startTimeMax).to.equal(1439);
  });

  describe('when the time changes, the time below ', () => {
    beforeEach(() => {
      rootComponent = TestUtils.renderIntoDocument(
        <IntlProvider locale={i18n.getLocale()}>
          <TimeRangeSlider
            startTimeMin={1380}
            startTimeMax={1439}
            handleFilterChange={mockFunction}
            ref={(c) => {
              renderedComponent = c.refs.wrappedInstance;
            }
          }
          />
        </IntlProvider>
      );
    });
    it('start time shows as "11::00 PM"', () => {
      const startTime = TestUtils.scryRenderedDOMComponentsWithClass(
        renderedComponent,
        'slider-value'
      );
      expect(startTime[0].textContent).to.equal('11:00 PM');
    });
    it('end time shows as "11::59 PM"', () => {
      const startTime = TestUtils.scryRenderedDOMComponentsWithClass(
        renderedComponent,
        'slider-value'
      );
      expect(startTime[1].textContent).to.equal('11:59 PM');
    });

    it('the slider filler is working', () => {
      const sliderFiller = TestUtils.findRenderedDOMComponentWithClass(
        renderedComponent,
        'slider-fill'
      );
      expect(sliderFiller.getAttribute('data-slider-fill')).to.equal('true');
    });
  });

  describe('when the locale changes to GB, the time below ', () => {
    beforeEach(() => {
      rootComponent = TestUtils.renderIntoDocument(
        <IntlProvider locale={'en-GB'}>
          <TimeRangeSlider
            startTimeMin={0}
            startTimeMax={1439}
            handleFilterChange={mockFunction}
            ref={(c) => {
              renderedComponent = c.refs.wrappedInstance;
              return renderedComponent;
            }
          }
          />
        </IntlProvider>
      );
    });
    it('start time shows as "00::00"', () => {
      const startTime = TestUtils.scryRenderedDOMComponentsWithClass(
        renderedComponent,
        'slider-value'
      );
      expect(startTime[0].textContent).to.equal('00:00');
    });
    it('end time shows as "23::59"', () => {
      const startTime = TestUtils.scryRenderedDOMComponentsWithClass(
        renderedComponent,
        'slider-value'
      );
      expect(startTime[1].textContent).to.equal('23:59');
    });
  });
});
```

**Comment by caridy** [comment]
This was fixed by 88fe47849422ce711aba86173a57e19b0b304d82
